sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs

```diff
@@ -199,6 +199,50 @@ public new AppendBlobClient WithSnapshot(string snapshot)
             return new AppendBlobClient(builder.ToUri(), this.Pipeline);
         }
 
+        /// <summary>
+        /// The <see cref="Create"/> operation creates a new 0-length
+        /// append blob.  The content of any existing blob is overwritten with
+        /// the newly initialized append blob.  To add content to the append
+        /// blob, call the <see cref="AppendBlock"/> operation.
+        /// 
+        /// For more information, see <see href="https://docs.microsoft.com/rest/api/storageservices/put-blob" />.
+        /// </summary>
+        /// <param name="httpHeaders">
+        /// Optional standard HTTP header properties that can be set for the
+        /// new append blob.
+        /// </param>
+        /// <param name="metadata">
+        /// Optional custom metadata to set for this append blob.
+        /// </param>
+        /// <param name="accessConditions">
+        /// Optional <see cref="AppendBlobAccessConditions"/> to add
+        /// conditions on the creation of this new append blob.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{BlobContentInfo}"/> describing the
+        /// newly created append blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response<BlobContentInfo> Create(
+            BlobHttpHeaders? httpHeaders = default,
```

Comment:
IMO I'd rather see the sync APIs as a separate extensions library, to make it easier for us to get rid of them at some point without necessarily invoking a breaking change for async customers.

Reply:
I know this was debated a lot by the architects.  The decision was made to support sync APIs across the board and at some point, once we proper sync support in the transport layer, this will be a fully sync call instead of just sync-over-async like extension methods would have to be.

Reply:
Is there any thought that we'd be able to wean folks off the sync APIs?  I know history isn't favorable toward that, but...

Reply:
(The extension methods I had in mind wouldn't be like "EnsureCompleted(this Task)" but "Foo(this AppendBlobClient)" to complement "AppendBlobClient.FooAsync()"

Reply:
Unfortunately not - too many customers want sync APIs.